virttest/utils_net.py

```diff
@@ -1050,15 +1050,15 @@ def get_structure(self):
         result = dict()
         for br_iface in os.listdir(sysfs_path):
             br_iface_path = os.path.join(sysfs_path, br_iface)
-            if os.path.isdir(br_iface_path):
-                try:
-                    if "bridge" not in os.listdir(br_iface_path):
-                        continue
-                except OSError as e:
-                    if e.errno == errno.ENOENT:
-                        continue
-                    else:
-                        raise e
+            try:
+                if (not os.path.isdir(br_iface_path) or
+                        "bridge" not in os.listdir(br_iface_path)):
+                    continue
+            except OSError as e:
+                if e.errno == errno.ENOENT:
+                    continue
+                else:
+                    raise e
 
```

Comment:
I have more or less the same comment as in the merged PR of this:

We encounter similar errors in the blocks below possibly due to the same reasons like:

```
20:17:00 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1100, in port_to_br
20:17:00 ERROR|     if port_name in self.get_structure()[br]['iface']:
20:17:00 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1065, in get_structure
20:17:00 ERROR|     with open(stp_state_path, "r") as stp_state_file:
20:17:00 ERROR| FileNotFoundError: [Errno 2] No such file or directory: '/sys/class/net/t0-W6kgkf/bridge/stp_state'
```

and sometimes

```

2019-07-09 16:27:01,051 stacktrace       L0045 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1100, in port_to_br
2019-07-09 16:27:01,051 stacktrace       L0045 ERROR|     if port_name in self.get_structure()[br]['iface']:
2019-07-09 16:27:01,051 stacktrace       L0045 ERROR|   File "/usr/lib/python3.7/site-packages/virttest/utils_net.py", line 1066, in get_structure
2019-07-09 16:27:01,052 stacktrace       L0045 ERROR|     stp_state = int(stp_state_file.read().strip())
2019-07-09 16:27:01,052 stacktrace       L0045 ERROR| OSError: [Errno 9] Bad file descriptor
```

Can we put this entire part in a try block instead?

As you can see the bridge might disappear also after the check above with the errors above collected at line 1065 and 1066 as well as possibly all other lines until the end of this function.

Reply:
In fact, in most cases, the following three lines are enough:
```python
                if (not os.path.isdir(br_iface_path) or
                        "bridge" not in os.listdir(br_iface_path)):
                    continue
```
An extreme case is that the interface directory is deleted between the `os.path.isdir(br_iface_path)` and `"bridge" not in os.listdir(br_iface_path)` steps. So need to `except OSError` and analyze.

Finally, `result[br_iface] = dict()` will be a correct dictionary, the following two errors do not occur:
```
20:17:00 ERROR| FileNotFoundError: [Errno 2] No such file or directory: '/sys/class/net/t0-W6kgkf/bridge/stp_state'
2019-07-09 16:27:01,052 stacktrace       L0045 ERROR|     stp_state = int(stp_state_file.read().strip())
```

Reply:
I will apply your patch and test to confirm that this actually improves our failure rate where we get the two errors above about 1/500 tests which is a lot as we have about 750 or so tests (read: error of this type occurs on almost every run we do). I will report back.

Reply:
So far I didn't encounter the problem even though I am not convinced the solution will cover 100% of the possibilities. Still, LGTM